docs/_posts/testing_a_tricky_edge_case_in_a_tuf_client.md

```diff
@@ -0,0 +1,90 @@
+  "version": 2
+ }}
+</code></pre>
+
+## Edge cases are a norm when we talk about security
+
+The TUF spec says the following:
+“Record the time at which the update began as the fixed update start time. Time is fixed at the beginning of the update workflow to allow an application using TUF to effectively pause time, in order to ensure that metadata which has a valid expiration time at the beginning of an update does not fail an expiration check later in the update workflow.”
+
+Now imagine that we have performed an update and stored metadata in cache. And the locally stored timestamp/snapshot has expired. But we still need it to perform an update from remote by verifying the signatures and we need to use the expired timestamp. Here the repository simulator can do a great job, as we’d like to somehow update expiry without explicitly touching the file system from the tests.
+
+We can play with versions and expiry to verify that this scenario not explicitly mentioned in the spec works correctly and safely. By using the simulator, we can set the timestamp expiry one week ahead. Then on the very first day we download, verify, and load metadata for the [top-level roles](https://theupdateframework.github.io/specification/latest/#roles-and-pki) following the specified order and implementing all the checks required in the TUF client workflow (which we simply call a refresh). Then we bump [snapshot](https://theupdateframework.github.io/specification/latest/#update-snapshot) and [targets](https://theupdateframework.github.io/specification/latest/#targets) versions to v2 on the same day and set v2 expiry dates three weeks ahead. If we travel in time somewhere between day 7 and day 21, we’d like to perform a successful refresh with the expired locally cached timestamp and the final version of the snapshot and targets roles to be v2.
+This is a not so obvious use-case to keep in mind when thinking about updates. You can see how it looks in practice in the [reference implementation](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_metadata).
+
+Now let’s see if a rollback attack protection can be performed when the local timestamp has expired. In this case we need at least two timestamp and snapshot versions, an expired older version of timestamp, and a verification that a rollback check is performed with the old version.
+For a timestamp rollback, the case is pretty similar to the use of  expired metadata. We can set timestamp v1 expiry one week ahead, perform refresh on the very first day, publish timestamp v2 with expiry three weeks ahead and perform a refresh somewhere between day 7 and day 21. All we need is to verify that rollback check uses the expired timestamp v1. (For reference, see the implementation [example](https://github.com/theupdateframework/python-tuf/blob/develop/tests/test_updater_top_level_update.py#:~:text=test_expired_timestamp_version_rollback)).
```

Comment:
Maybe you add a new space to actually divide the two paragraphs because markdown merges them as if there is no newline between lines number 83 and 84?

Reply:
Again as above, I recommend using numbers and writing them in separate steps.